### PR TITLE
Finish the task creation verification

### PR DESCRIPTION
**Pull request by lenak25**
When a worker receives a new task to compute, as part of its verification, the task inputs hash needs to be compared to what appears in enigma contract. 
After the hashing mechanism was decided upon and implemented in the contract side, the P2P should be aligned.

### CODE REVIEW

**Comment by lenak25** [comment]
@elichai please review

File: src/common/cryptography.js [hunk 1]
```diff
@@ -32,6 +35,24 @@ module.exports.toBN = (value) => {
...
 
+
+/**
+ * Generate a hash of all inputs
+ * The Enigma contract uses the same logic to generate a matching taskId
+ *
+ * @param {array} inputsArray
+ * @return {string} hash of inputs
+ */
+module.exports.hashArray = (inputsArray) => {
```
Comment (elichai) [inline src/common/cryptography.js]: so this accepts hex strings, right?

File: src/common/cryptography.js [hunk 2]
```diff
@@ -32,6 +35,24 @@ module.exports.toBN = (value) => {
...
+  let hexStr = '';
+  for (let e of inputsArray) {
+    e = utils.remove0x(e);
+    // since the inputs are in hex string, they are twice as long as their bytes
+    hexStr += (new BN(e.length/2).toString(16, 16)) + e;
```
Comment (elichai) [inline src/common/cryptography.js]: I don't like that it doesn't say anything about endianess, who decides that?

Comment (elichai) [reply src/common/cryptography.js]: We can be more explicit about this by doing something like:
```
hexStr += (new BN(e.length/2).toBuffer('be', 8).toString('hex')) + e;
```
(the 'be' means BigEndian)

but it looks uglier, i'll try to look more through the docs

**Pull request merged**